cvat/apps/dataset_manager/views.py

```diff
@@ -58,6 +60,42 @@ def get_export_cache_ttl(db_instance: str | Project | Task | Job) -> timedelta:
 
     return TTL_CONSTS[db_instance.lower()]
 
+def _retry_current_rq_job(time_delta: timedelta) -> rq.job.Job:
+    # TODO: implement using retries once we move from rq_scheduler to builtin RQ scheduler
+    # for better reliability and error reporting
+
+    # This implementation can potentially lead to 2 jobs with the same name running in parallel,
+    # if the retry is enqueued immediately.
+    assert time_delta.total_seconds() > 0
+
+    current_rq_job = rq.get_current_job()
+
+    def _patched_retry(*_1, **_2):
+        scheduler: Scheduler = django_rq.get_scheduler(
+            settings.CVAT_QUEUES.EXPORT_DATA.value
+        )
+
+        user_id = current_rq_job.meta.get('user', {}).get('id') or -1
+
+        with get_rq_lock_by_user(settings.CVAT_QUEUES.EXPORT_DATA.value, user_id):
+            scheduler.enqueue_in(
+                time_delta,
+                current_rq_job.func,
+                *current_rq_job.args,
+                **current_rq_job.kwargs,
+                job_id=current_rq_job.id,
+                meta=current_rq_job.meta,
+                depends_on=current_rq_job.dependency_ids,
```

Comment:
@zhiltsov-max, Probably, there are several bugs:
1) `dependency_ids` can be outdated at the moment when this job will be enqueued, so one user can clog a queue with his jobs (despite `ONE_RUNNING_JOB_IN_QUEUE_PER_USER`==True)
2) let's imagine a situation when:
there were 3 rq jobs: `J1` <- `J2(allow_dependency_failures=True)` <- `J3(allow_dependency_failures=True)`
`J1` was executed successfully -> `J2` was started -> `J1` was deleted by CVAT code after downloading a file -> `J2` was scheduled with allow_dependency_failures = None (because it is not copied in the current implementation), -> `J1` was recreated after one more request from a user and now `J1` depends on the `J3`. What will be at the end? Will J3 be enqueued after `J2` is placed to the scheduled job registry?